### ISSUE TITLE
syntax error in downgraded package

### ISSUE THREAD

**Issue by bloep** [symplify/phpstan-rules#60]
The new downgraded release https://github.com/symplify/phpstan-rules/releases/tag/12.0.3.72 throws an error with php 7.2

```
PHP Parse error:  syntax error, unexpected end of file, expecting variable (T_VARIABLE) or heredoc end (T_END_HEREDOC) or ${ (T_DOLLAR_OPEN_CURLY_BRACES) or {$ (T_CURLY_OPEN) in /home/runner/work/project/vendor/symplify/phpstan-rules/src/Rules/NoReturnSetterMethodRule.php on line 129
Parse error: syntax error, unexpected end of file, expecting variable (T_VARIABLE) or heredoc end (T_END_HEREDOC) or ${ (T_DOLLAR_OPEN_CURLY_BRACES) or {$ (T_CURLY_OPEN) in /home/runner/work/project/vendor/symplify/phpstan-rules/src/Rules/NoReturnSetterMethodRule.php on line 129
```
----

my tests have confirmed that it has to do with the formatting of the heredoc.

original rule:
https://github.com/symplify/phpstan-rules/blob/c584728fa538b0bc9bb3582ca3bf45b9872b1dc4/src/Rules/NoReturnSetterMethodRule.php#L95-L97

php7.2 downgraded:
https://github.com/symplify/phpstan-rules/blob/b192fada83caaabfcc78ed553c34eead67501a36/src/Rules/NoReturnSetterMethodRule.php#L99

**Comment by samsonasik** [comment]
@TomasVotruba The downgrade process is using rector 0.16.0 versioning, see https://github.com/symplify/phpstan-rules/actions/runs/5222642850/jobs/9428399337#step:4:50

so it possibly bug on rector 0.16

**Comment by samsonasik** [comment]
I tried with rector 0.17.1, it seems the diff applied without removing heredoc posisiton:

```diff
-final class NoReturnSetterMethodRule implements Rule, DocumentedRuleInterface
+final class NoReturnSetterMethodRule implements Rule
 {
     /**
+     * @readonly
+     * @var \Symplify\PHPStanRules\NodeFinder\TypeAwareNodeFinder
+     */
+    private $typeAwareNodeFinder;
+    /**
      * @var string
      */
     public const ERROR_MESSAGE = 'Setter method cannot return anything, only set value';
@@ @@
      */
     private const SETTER_START_REGEX = '#^set[A-Z]#';

-    public function __construct(
-        private readonly TypeAwareNodeFinder $typeAwareNodeFinder
-    ) {
+    public function __construct(TypeAwareNodeFinder $typeAwareNodeFinder)
+    {
+        $this->typeAwareNodeFinder = $typeAwareNodeFinder;
     }

     /**
    ----------- end diff -----------
```

I will try create new release for it with new rector 0.17.1.

**Comment by samsonasik** [comment]
@bloep Fixed in latest symplify/phpstan-rules 12.0.4.72 

https://github.com/symplify/phpstan-rules/blob/ff0336a3b750c8ad036fb57ee663d775355bc463/src/Rules/NoReturnSetterMethodRule.php#L90-L111
